### ISSUE TITLE
AUC metric

### ISSUE THREAD

**Issue by MaxHalford** [creme-ml/creme#84]
AUC can't be computed online without storing all the observed true and predicted values, but an approximate implementation exists. See [tensorflow.metrics.aux](https://www.tensorflow.org/api_docs/python/tf/metrics/auc).

**Comment by MaxHalford** [comment]
@AdilZouitine I just pushed a simple implementation, I compared it with `sklearn` and the scores were very close. However, just like Tensorflow, it doesn't work very well if the predicted probabilities are too close to 0 or 1, but this is something we can fix later. The current implementation uses thresholds that are generated uniformly between 0 and 1, but we could use thresholds that are denser after 0 and before 1, and less dense around 0.5.
